drv/lpc55-update-server/src/main.rs

```diff
@@ -224,6 +228,29 @@ impl idl::InOrderUpdateImpl for ServerImpl<'_> {
         RotBootState::load_from_addr(addr).map_err(|e| e.into())
     }
 
+    fn rot_boot_info(
+        &mut self,
+        _: &RecvMessage,
+    ) -> Result<RotBootInfo, RequestError<UpdateError>> {
+        // Safety: Data is published by pre-kernel main
+        let addr = unsafe { BOOTSTATE.assume_init_ref() };
+        let boot_state = RotBootState::load_from_addr(addr)
+            .map_err(|_| UpdateError::MissingHandoffData)?;
+        let (
+            persistent_boot_preference,
+            pending_persistent_boot_preference,
+            transient_boot_preference,
+        ) = self.boot_preferences()?;
+
+        let info = RotBootInfo {
```

Comment:
Nit: you could return this directly instead of assigning it (`Ok(RotBootInfo { ... })`)

Reply:
I know. I think I like the local better :)